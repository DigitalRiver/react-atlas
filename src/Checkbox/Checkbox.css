@import '../styles/variables.css';

.disabled {
    composes: default-text from '../styles.css';
}

.relative {
    position:relative;
}

.padding {
    padding-bottom: 5px;
}

.input {
    composes: relative;
    top: 1px;
}

.checkbox {
  composes: rounded from '../styles.css';
  composes: default-border from "../styles.css";
  composes: relative;
  position: absolute;
  top: 0px;
  width: 16px;
  height: 16px;
  cursor: pointer;
}

.disabled {
  opacity: 0.65;
  composes: cursor-not-allowed from '../styles.css'; 
  pointer-events: none;
}

.checked {
  composes: checkbox;
  composes: bg-primary from '../styles.css';
}

.hidden {
  display: none;
  visibility: hidden;
}

.focus {
  box-shadow: 0 0 6px var(--primary)
}

.checkmark {
  composes: relative;
  top: 0px;
  left: 4px;
  width: 4px;
  height: 10px;
  transform: rotate(45deg);
  border: solid var(--white);
  border-width: 0 3px 3px 0;    
}

.error {
  border: solid 1px var(--error);
}

/* Display checkbox with out the checkbox selected.  */
.not_checked {
  composes: checkbox;
  background: -webkit-linear-gradient(#fff, #e6e6e6); /* For Safari 5.1 to 6.0 */
  background: -o-linear-gradient(#fff, #e6e6e6); /* For Opera 11.1 to 12.0 */
  background: -moz-linear-gradient(#fff, #e6e6e6); /* For Firefox 3.6 to 15 */
  background: linear-gradient(#fff, #e6e6e6); /* Standard syntax */
}

.label {
  composes: default-font from '../styles.css';
  cursor: pointer;
  white-space: nowrap;
  padding-right:5px;
}

/* label text */
.label_left {
  composes: label;
  padding-right: 2px;
  margin-left: 0px;
}

.error_text {
  composes: default-font from '../styles.css';
  composes: error-text from '../styles.css';
}

.error_message {
  composes: error_text;
  margin-left:23px;
}

.inline_block {
  display: inline-block;
}

.checkbox_padding {
  padding: 10px 10px 0 0;
}

.inline_padding {
  padding: 2px 10px 0 0;
}

.float_left {
  float:left;
  padding-right: 3px;
}

.float_right {
  float:right;
  padding-left: 3px;
  position: relative;
}

.opacity {
  /*this needs to stay here to avoid display issues with those modules in IE*/
  opacity: 0;
}

.required {
  composes: secondary-text from '../styles.css';
}

.required_error {
  composes: error-text from '../styles.css';
}