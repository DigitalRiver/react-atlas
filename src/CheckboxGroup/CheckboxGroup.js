import React, { cloneElement } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./CheckboxGroup.css";

export class CheckboxGroup extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      "totalChecked": this.initialChecked(),
      "groupError": false
    };
  }

  // Loop through child components to determine how many <Checkbox /> components are checked by default
  initialChecked() {
    let newChecked = 0;
    React.Children.map(this.props.children, child => {
      if (child.props.checked) {
        newChecked++;
      }
    });
    return newChecked;
  }

  // When child <Checkbox /> component changes state, update "totalChecked" value and verify against min or max props value
  handleChange = data => {
    const newChecked = data.checked
      ? this.state.totalChecked + 1
      : this.state.totalChecked - 1;
    if (data.checked) {
      this.setState({ "totalChecked": newChecked });
    } else {
      this.setState({ "totalChecked": newChecked });
    }
    if (
      this.props.max && newChecked > this.props.max ||
      this.props.min && newChecked < this.props.min
    ) {
      this.setState({ "groupError": true });
    } else {
      this.setState({ "groupError": false });
    }

    if (typeof this.props.onChange !== "undefined") {
      this.props.onChange(data);
    }
  };

  // If min or max props are set and the correct number of checkboxes aren't checked, display the appropriate error message
  maxMinMessage = () => {
    let fullMessage;
    if (this.props.max && this.state.totalChecked > this.props.max) {
      if (this.props.limitMessage) {
        fullMessage = this.props.limitMessage.replace("{0}", this.props.max);
        return fullMessage;
      } else {
        return `Please select no more than ${
          this.props.max
        } of the options below`;
      }
    } else if (this.props.min && this.state.totalChecked < this.props.min) {
      if (this.props.limitMessage) {
        fullMessage = this.props.limitMessage.replace("{0}", this.props.min);
        return fullMessage;
      } else {
        return (
          "Please select at least " + this.props.min + " of the options below"
        );
      }
    } else {
      return false;
    }
  };

  render() {
    const {
      className,
      children,
      name,
      inline,
      inlineChildren,
      title,
      style
    } = this.props;
    const checkboxGroupStyles = cx({
      "checkboxGroup": !inline,
      "inline": inline
    });
    return (
      <div
        style={style}
        className={cx(className)}
        styleName={checkboxGroupStyles}
      >
        {title && 
          <div styleName={"header"}>
            <span styleName={"headerFont"}>{title}</span>
            {this.state.groupError && 
              <span styleName={"error_message"}>{this.maxMinMessage()}</span>
            }
          </div>
        }
        {React.Children.map(children, child => {
          child = cloneElement(child, {
            "inline": inlineChildren,
            "name": name,
            "onChange": this.handleChange,
            "groupError": this.state.groupError
          });
          return child;
        })}
      </div>
    );
  }
}

CheckboxGroup.propTypes = {
  /**
   * Anything that can be in a CheckboxGroup. Typically only includes Checkbox components and a header.
   * @examples '<CheckboxGroup><Checkbox/><Checkbox/></CheckboxGroup>'
   */
  "children": PropTypes.node.isRequired,

  /** An object, array, or string of CSS classes to apply to CheckboxGroup.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * When true, will display the CheckboxGroup inline.
   * @examples '<CheckboxGroup inline></CheckboxGroup>'
   */
  "inline": PropTypes.bool,

  /**
   * When true, will display child Checkbox components inline.
   * @examples '<CheckboxGroup inlineChildren></CheckboxGroup>'
   */
  "inlineChildren": PropTypes.bool,

  /**
   * Will define a custom message to display if either min or max properties are not met. {0} can be used in the message to be replaced by the appropriate min or max property.
   * @examples '<CheckboxGroup min="3" limitMessage="Please select at least {0} of the checkboxes below."></CheckboxGroup>'
   */
  "limitMessage": PropTypes.string,

  /**
   * Will define the maximum number of checkboxes the user is allowed to select before proceeding.
   * @examples '<CheckboxGroup max="3"></CheckboxGroup>'
   */
  "max": PropTypes.number,

  /**
   * Will define the minimum number of checkboxes the user must select in order to proceed.
   * @examples '<CheckboxGroup min="3"></CheckboxGroup>'
   */
  "min": PropTypes.number,

  /**
   * Form name for the element. This will set all Checkbox children the same form name.
   * @examples '<CheckboxGroup name="test"></CheckboxGroup>'
   */
  "name": PropTypes.string,

  /**
   * An optional callback that will be fired when any checkbox in the
   * CheckboxGroup changes value. The callback has four arguments.
   * function(value, event, isValid, checked).
   */
  "onChange": PropTypes.func,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,

  /**
   * Text that will be displayed above the checkboxes in the CheckboxGroup.
   * @examples '<CheckboxGroup title="Options"></CheckboxGroup>'
   */
  "title": PropTypes.string
};

CheckboxGroup.defaultProps = {
  "inline": false
};

export default CSSModules(CheckboxGroup, styles, { "allowMultiple": true });
