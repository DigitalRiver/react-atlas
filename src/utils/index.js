import events from "./events";
import prefixer from "./prefixer";
// import time from './time';
import utils from "./utils";

export { events, prefixer, utils };
