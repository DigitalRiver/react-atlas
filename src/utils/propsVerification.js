export function verifyPropsDefaultValue(component, expectedProps) {
  //Compares each property value of the component with the expected value

  let result = true;
  let partialResult;

  for (let [compPropName, compPropValue] of expectedProps.entries()) {
    partialResult = component.props()[compPropName] === compPropValue;

    if (!partialResult) {
      console.log(
        "[" +
          compPropName +
          "] -> Got: " +
          component.props()[compPropName] +
          " - expected: " +
          compPropValue
      );
      result = false;
    }
  }
  return result;
}
