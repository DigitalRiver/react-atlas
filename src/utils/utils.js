import React from "react";
import messages from "./messages.js";

export default {
  angleFromPositions(cx, cy, ex, ey) {
    const theta = Math.atan2(ey - cy, ex - cx) + Math.PI / 2;
    return theta * 180 / Math.PI;
  },
  angle360FromPositions(cx, cy, ex, ey) {
    const angle = this.angleFromPositions(cx, cy, ex, ey);
    return angle < 0 ? 360 + angle : angle;
  },
  range(start = 0, stop = null, step = 1) {
    let [_start, _stop] = [0, start];
    if (stop !== null) {
      [_start, _stop] = [start, stop];
    }
    const length = Math.max(Math.ceil((_stop - _start) / step), 0);
    const range = Array(length);

    for (let idx = 0; idx < length; idx++, _start += step) {
      range[idx] = _start;
    }

    return range;
  },
  round(number, decimals) {
    if (!isNaN(parseFloat(number)) && isFinite(number)) {
      const decimalPower = Math.pow(10, decimals);
      return Math.round(parseFloat(number) * decimalPower) / decimalPower;
    }
    return NaN;
  },
  getViewport() {
    return {
      "height": window.innerHeight || document.documentElement.offsetHeight,
      "width": window.innerWidth || document.documentElement.offsetWidth
    };
  },
  cloneObject(object) {
    return JSON.parse(JSON.stringify(object));
  },
  inputTypeForPrototype(prototype) {
    if (prototype === Date) {
      return "date";
    }
    if (prototype === Number) {
      return "number";
    }
    if (prototype === Boolean) {
      return "checkbox";
    }
    return "text";
  },
  prepareValueForInput(value, type) {
    if (type === "date") {
      return new Date(value).toISOString().slice(0, 10);
    }
    if (type === "checkbox") {
      return value ? "on" : null;
    }
    return value;
  },
  getSelection(elem) {
    let start, end, rangeElem, clone;

    if (typeof elem.selectionStart !== "undefined") {
      start = elem.selectionStart;
      end = elem.selectionEnd;
    } else {
      try {
        elem.focus();
        rangeElem = elem.createTextRange();
        clone = rangeElem.duplicate();

        rangeElem.moveToBookmark(
          document.selection.createRange().getBookmark()
        );
        clone.setEndPoint("EndToStart", rangeElem);

        start = clone.text.length;
        end = start + rangeElem.text.length;
      } catch (e) {
        /* not focused or not visible */
      }
    }

    return { start, end };
  },
  setSelection(elem, selection) {
    let rangeElem;

    try {
      if (typeof elem.selectionStart !== "undefined") {
        elem.focus();
        elem.setSelectionRange(selection.start, selection.end);
      } else {
        elem.focus();
        rangeElem = elem.createTextRange();
        rangeElem.collapse(true);
        rangeElem.moveStart("character", selection.start);
        rangeElem.moveEnd("character", selection.end - selection.start);
        rangeElem.select();
      }
    } catch (e) {
      /* not focused or not visible */
    }
  },
  getComponentName(comp) {
    if (comp.type) {
      if (
        Object.keys(comp.type).length === 0 &&
        comp.type.constructor === Object
      ) {
        return null;
      }
      return comp.type.displayName || comp.type.name || comp.type || null;
    }
    return null;
  },
  filter(object, excludes) {
    return Object.keys(object)
      .filter(obj => !excludes[obj])
      .reduce((obj, key) => {
        obj[key] = object[key];
        return obj;
      }, {});
  },
  getRequiredText(reqText) {
    return <span styleName={"error_text"}>{reqText}</span>;
  },
  getErrorMessage(errorMessage) {
    return <div styleName={"error_message"}>{errorMessage}</div>;
  },
  validate(inputValue, validator, propStatus, propMessage, required, event) {
    let status = null;
    let message = null;
    /* Execute custom validator and change state and error messages accordingly */
    if (typeof validator === "function") {
      let validationObject = validator(event, inputValue);
      if (typeof validationObject !== "undefined") {
        if (
          typeof validationObject === "boolean" &&
          validationObject === true
        ) {
          status = propStatus;
          message = propMessage;
        } else if (
          typeof validationObject === "boolean" &&
          validationObject === false
        ) {
          status = "error";
        } else {
          status = validationObject.status;
          message = validationObject.message;
        }
      }
    } else if (
      (typeof inputValue === "undefined" || inputValue === "") &&
      (required || typeof required === "string" && required === "")
    ) {
      /* If the field is required, and it has no value, change state and display error message */
      message = messages.requiredMessage;
      status = "error";
    }
    return { "status": status, "message": message };
  }
};
