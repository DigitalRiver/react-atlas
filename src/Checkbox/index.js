import Checkbox from "./Checkbox.js";
export { Checkbox };
