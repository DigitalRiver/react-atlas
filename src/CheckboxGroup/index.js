import CheckboxGroup from "./CheckboxGroup";
export { CheckboxGroup };
