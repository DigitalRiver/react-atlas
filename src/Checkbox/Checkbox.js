import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { utils } from "../utils";
import messages from "../utils/messages.js";
import CSSModules from "react-css-modules";
import styles from "./Checkbox.css";
import { Text } from "../Text";
import blacklist from "blacklist";

/**
 * Simple component for a basic checkbox
 */
export class Checkbox extends React.PureComponent {
  constructor(props) {
    super(props);

    // Initial state
    this.state = {
      "checked": this.props.checked || false,
      "status": props.status || null,
      "message": props.message || null,
      "focus": false
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.checked !== this.props.checked) {
      this.setState({
        "checked": nextProps.checked || false
      });
    }
  }
  // Handles new checkbox clicks and sets value and checked status of hidden input
  _clickHandler = event => {
    if (this.props.disabled) {
      return false;
    }
    event.persist();
    if (typeof this.props.onBeforeChange !== "undefined") {
      let result = this.props.onBeforeChange(this.state.checked);
      if (result === false) {
        return;
      }
    }
    const validationObject = this._validationHandler(this.props.valid);
    this.setState(
      {
        "checked": !this.state.checked,
        "status": validationObject.status,
        "message": validationObject.message
      },
      function() {
        const data = {
          "value": this.props.value,
          "status": this.state.status,
          "checked": this.state.checked
        };
        /* Check if onClick has been passed, if so call it. */
        if (typeof this.props.onClick !== "undefined") {
          this.props.onClick(data);
        }
        /* Check if onChange has been passed, if so call it. */
        if (typeof this.props.onChange !== "undefined") {
          this.props.onChange(data);
        }
      }
    );
  };

  // Added to remove console warning for controlled/uncontrolled component
  _handleChange = () => {
    return false;
  };

  _getStatus = preState => {
    let status = "error";
    if (
      preState &&
      (this.props.required && !this.state.checked || !this.props.required)
    ) {
      status = null;
    } else if (
      !preState &&
      (this.props.required && this.state.checked || !this.props.required)
    ) {
      status = null;
    }
    return status;
  };

  _validationHandler = callback => {
    // If custom validation callback is provided set validationObject with response, otherwise check if required
    const validationObject = callback
      ? callback(event, !this.state.checked)
      : {
          "status": this._getStatus(true),
          "message": messages.requiredMessage
        };
    return validationObject;
  };

  // Used from a parent Form component to validate only
  _formValidate = () => {
    const validationObject = this.props.valid
      ? this.props.valid(event, !this.state.checked)
      : {
          "status": this._getStatus(false),
          "message": messages.requiredMessage
        };
    this.setState({
      "status": validationObject.status,
      "message": validationObject.message
    });
    return validationObject;
  };

  render() {
    const {
      className,
      disabled,
      groupError,
      hidden,
      id,
      inline,
      label,
      labelPosition,
      required,
      status,
      style,
      title,
      ...others
    } = this.props;

    // Declaring the following variables so they don't get passed to the input through the prop spread.
    const othersFiltered = blacklist(others, "onBeforeChange", "valid");

    const inlineCheckbox = cx({
      "inline_block": inline,
      "checkbox_padding": !inline,
      "inline_padding": inline,
      hidden
    });
    const labelStyle = cx({
      "label": labelPosition !== "left",
      "label_left": labelPosition === "left"
    });
    const checkboxDisplay =
      labelPosition === "left" ? "float_right" : "float_left";
    const title_label = title ? title : label;
    let disabledClass = cx({
      disabled,
      "inline_block": true,
      "relative": true,
      "padding": !inline
    });
    const error = groupError || this.state.status === "error";
    let checkboxClass = cx({
      "checked": this.state.checked,
      "error": error,
      "not_checked": !this.state.checked,
      "focus": this.state.focus
    });

    // Gets the appropriate jsx to render a "required" identifier next to the Checkbox.
    const reqText = typeof required === "string" ? required : "*";

    // Gets the appropriate jsx to render an error message below the Checkbox.
    const errorMessage =
      this.state.status === "error"
        ? utils.getErrorMessage(this.state.message)
        : null;

    const requiredClasses = cx({
      "required": true,
      "required_error": status === "error"
    });

    return (
      <div
        onClick={this._clickHandler}
        styleName={inlineCheckbox}
        style={style}
        className={cx(className)}
      >
        <div styleName={disabledClass}>
          {label && 
            <label styleName={labelStyle} title={title_label} htmlFor={id}>
              {label}
            </label>
          }
          <div styleName={checkboxDisplay}>
            <input
              {...othersFiltered}
              label={label}
              type="checkbox"
              disabled={disabled}
              checked={this.state.checked}
              required={required}
              hidden={hidden}
              id={id}
              onChange={this._handleChange}
              styleName={cx("opacity")}
            />
            <div styleName={checkboxClass}>
              {this.state.checked && <div styleName={"checkmark"} />}
            </div>
          </div>
          {required &&
            required !== "" && 
              <span styleName={requiredClasses}>
                {" "}
                <Text>{reqText}</Text>
              </span>
            }
        </div>
        {errorMessage}
      </div>
    );
  }
}

Checkbox.propTypes = {
  /**
   * When true, Checkbox will be checked on load.
   */
  "checked": PropTypes.bool,

  /** An object, array, or string of CSS classes to apply to Checkbox.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * If included, checkbox is disabled
   * @examples <Checkbox disabled />, <Checkbox disabled={true} />
   */
  "disabled": PropTypes.bool,

  /**
   * For use when an error state has been passed down from the parent CheckboxGroup.
   */
  "groupError": PropTypes.bool,
  /**
   * Determines if the Checkbox is hidden.
   * @examples '<Checkbox hidden/>'
   */
  "hidden": PropTypes.bool,

  /** Will set the html "id" property on the Checkbox.*/
  "id": PropTypes.string,

  /**
   * When true, will render the Checkbox and its label inline.
   */
  "inline": PropTypes.bool,

  /**
   * Text for Checkbox label.
   * @examples 'Some Label'
   */
  "label": PropTypes.string,

  /**
   * Will position the Checkbox label to the left or the right of the checkbox.
   */
  "labelPosition": PropTypes.string,

  /**
   * Function that will be executed before changing the "checked" state of the Checkbox.
   */
  "onBeforeChange": PropTypes.func,

  /**
   * Function that will be executed when the Checkbox state is changed.
   */
  "onChange": PropTypes.func,

  /**
   * Function that will be executed on click.
   */
  "onClick": PropTypes.func,

  /**
   * Sets the Checkbox as required. Will be validated in _validationHandler. Accepts a boolean or a string.
   * If a string is passed it will be displayed instead of the traditional * next to the Checkbox.
   * @examples '<TextField required/>'
   */
  "required": PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,

  /**
   * Text for Checkbox label title. (i.e. "alt-text" for checkboxes, useful for accessibility). If not provided, label text will be used.
   * @examples 'Some Title'
   */
  "title": PropTypes.string,

  /**
   * Used to pass a function for custom validation. Should return either true or false.
   */
  "valid": PropTypes.func,

  /**
   * The value of the Checkbox.
   */
  "value": PropTypes.string,

  /**
   * Sets the status of the Checkbox. Options are null, "success", "error", and "warning".
   * @examples '<Checkbox status="error" />'
   */
  "status": PropTypes.string,

  /**
   * Sets the status message of the Checkbox.
   * @examples '<Checkbox
   label="Required Example"
   valid={ function(checked){ return {status: "success", message: "Checkbox is required"} } }
   />'
   */
  "message": PropTypes.string
};

Checkbox.defaultProps = {
  "className": "",
  "disabled": false,
  "inline": false,
  "checked": false,
  "required": false
};

export default CSSModules(Checkbox, styles, { "allowMultiple": true });
