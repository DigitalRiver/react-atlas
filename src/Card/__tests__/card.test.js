import React from "react";
import { mount } from "enzyme";
import { Card } from "../index";
import { Checkbox } from "../../Checkbox/Checkbox";
import { CheckboxGroup } from "../../CheckboxGroup/CheckboxGroup";
import { verifyPropsDefaultValue } from "../../utils/propsVerification";

import renderer from "react-test-renderer";

describe("Test Card component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Card className={"name"} legend={"lalala"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test Card component: Basic Tests", () => {
  let cardComponent = null;

  it("Card - Basic properties test", function() {
    cardComponent = mount(<Card className={"name"} legend={"lalala"} />);
    const expected = new Map([["className", "name"], ["legend", "lalala"]]);
    expect(verifyPropsDefaultValue(cardComponent, expected)).toEqual(true);
  });

  it("Basic Accordion", () => {
    expect(
      cardComponent.exists(
        <Card>
          <div>Child</div>
        </Card>
      )
    ).toBe(true);
  });

  it("Contains No Child", () => {
    cardComponent = mount(<Card />);
    expect(cardComponent.props().children).toBeUndefined();
  });

  it("Card - Children test", function() {
    cardComponent = mount(
      <Card className={"name"} legend={"lalala"}>
        <div>Card Test</div>
      </Card>
    );
    expect(cardComponent.props()).toHaveProperty("children");
  });

  it("Test Standard Fieldset", function() {
    cardComponent = mount(
      <Card legend="Fieldset Example" standardFieldset>
        <CheckboxGroup title="Checkbox Group" name="checkboxGroup">
          <Checkbox label="Checkbox 1" checked />
          <Checkbox label="Checkbox 2" />
          <Checkbox label="Checkbox 3" />
        </CheckboxGroup>
      </Card>
    );
    expect(cardComponent.props().standardFieldset).toBe(true);
  });

  it("Card - Image prop test", function() {
    cardComponent = mount(
      <Card image="https://static1.squarespace.com/static/5369465be4b0507a1fd05af0/5397d14be4b06a5454f5633c/5397d15ae4b06a5454f56351/1404441173879/neo_001126-02.jpg?format=1500w" />
    );
    expect(cardComponent.props().image).toBe(
      "https://static1.squarespace.com/static/5369465be4b0507a1fd05af0/5397d14be4b06a5454f5633c/5397d15ae4b06a5454f56351/1404441173879/neo_001126-02.jpg?format=1500w"
    );
  });

  it("Test Title prop", () => {
    cardComponent = mount(<Card title="first">first</Card>);
    expect(cardComponent.props().title).toBe("first");
  });
});
