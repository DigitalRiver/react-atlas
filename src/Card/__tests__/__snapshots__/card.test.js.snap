// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component renders correctly 1`] = `
<div
  className="name card"
  style={undefined}
/>
`;
